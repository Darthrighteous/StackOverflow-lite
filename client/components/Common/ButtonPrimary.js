import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const ButtonPrimary = (props) => {
  const { link, label, colorBtn, colorTxt, onClick } = props;
  const btnStyle = {
    backgroundColor: colorBtn,
    color: colorTxt,
  };
  return(
    <NavLink activeStyle={{ display: 'none' }} to={link}>
      <button type="button" onClick={onClick} style={btnStyle}>
        {label}
      </button>
    </NavLink>
  );
};

ButtonPrimary.propTypes = {
  link: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  colorBtn: PropTypes.string.isRequired,
  colorTxt: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

ButtonPrimary.defaultProps = {
  onClick: null,
};

export default ButtonPrimary;