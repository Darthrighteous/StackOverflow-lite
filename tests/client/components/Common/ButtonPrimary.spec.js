import React from 'react';
import { shallow } from 'enzyme';

import ButtonPrimary from '../../../../client/components/Common/ButtonPrimary';

test('Primary button snapshot test', () => {
  const btnPrimary = shallow(
    <ButtonPrimary
      link="/test"
      label="test"
      colorBtn="#fff"
      colorTxt="#000"
    />
  );
  expect(btnPrimary).toMatchSnapshot();
});
