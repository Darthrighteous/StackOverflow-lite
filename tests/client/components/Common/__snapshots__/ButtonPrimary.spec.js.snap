// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary button snapshot test 1`] = `
<NavLink
  activeClassName="active"
  activeStyle={
    Object {
      "display": "none",
    }
  }
  aria-current="page"
  to="/test"
>
  <button
    onClick={null}
    style={
      Object {
        "backgroundColor": "#fff",
        "color": "#000",
      }
    }
    type="button"
  >
    test
  </button>
</NavLink>
`;
